client/utils/useStakingAPY.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useMemo, useState } from "react";
+import { SECONDS_IN_A_MONTH } from "constants/index";
+import { useStore } from "utils/store";
+import { getRewardsApy } from "./apy";
+import { ethers } from "ethers";
+
+const useStakingAPY = (amountStaked, duration) => {
+  const { contracts, totalBalances } = useStore();
+
+  const [loading, setLoading] = useState(true);
+  const [veOgvReceived, setVeOGVReceived] = useState(0);
+
+  const { totalSupplyVeOgvAdjusted } = totalBalances;
+
+  useEffect(() => {
+    if (!contracts.loaded) return;
+
+    let done = false;
```

Comment:
Not familiar with this pattern for how you are using `done`, what are you trying to solve? Just running the `go` function once?

Reply:
Just defensive programming. `go` is an async function. It's possible that the component might rerender before the function completes execution. If it does, then `setState` will throw a bunch of console errors because it's trying to update state of a detached component. When the component demounts, done is set to true and the state isn't updated in that case